File: Python/RearrangeArrayElements.py
# https://leetcode.com/problems/rearrange-array-elements-by-sign/description/

class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        n = len(nums)
        ans = [0] * n

        pos = 0
        neg = 1

        for i in range(n):
            if nums[i] > 0:
                ans[pos] = nums[i]
                pos += 2
            else:
                ans[neg] = nums[i]
                neg += 2
        
        return ans
    
if __name__ == "__main__":
    from typing import List
    size = int(input())
    nums = [int(t) for t in input().strip().split()]
    sol = Solution()
    ans = sol.rearrangeArray(nums)
    print(" ".join(map(str, ans)))

File: RearrangeArrayElements.java
// https://leetcode.com/problems/rearrange-array-elements-by-sign/

class RearrangeArrayElements{
    public int[] rearrangeArray(int[] nums) {
        int n = nums.length;
        int neg = 1;
        int pos = 0;
        int[] ans = new int[n];
        for(int i = 0; i < nums.length; i++){
            if(nums[i] > 0){
                ans[pos] = nums[i];
                pos = pos + 2;
            }
            else{
                ans[neg] = nums[i];
                neg = neg + 2;
            }
        }
        return ans;
    }

    public static void main(String[] args) {
        RearrangeArrayElements rearranger = new RearrangeArrayElements();
        int[] nums = {3, -2, 1, -5, 4, -6};
        int[] result = rearranger.rearrangeArray(nums);
        
        // Print the rearranged array
        for(int num : result){
            System.out.print(num + " ");
        }
    }
}